### ISSUE TITLE
memory allocation of 9897508864 bytes failed/148: binaryen-sys(build)[v0.8.0]

### ISSUE THREAD

**Issue by xhuanlee** [paritytech/cargo-contract#160]
error when install 0.8.0, but 0.7.1 is fine, any help will be appreciated.

```
   Compiling cargo_metadata v0.12.3
     Running `rustc --crate-name cargo_metadata --edition=2018 /root/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/cargo_metadata-0.12.3/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no --cfg 'feature="default"' -C metadata=2e6f6dea133bdcf5 -C extra-filename=-2e6f6dea133bdcf5 --out-dir /tmp/cargo-installvGQBzF/release/deps -L dependency=/tmp/cargo-installvGQBzF/release/deps --extern cargo_platform=/tmp/cargo-installvGQBzF/release/deps/libcargo_platform-14daf62d7e0647e6.rmeta --extern semver=/tmp/cargo-installvGQBzF/release/deps/libsemver-83ad9d90509556a4.rmeta --extern semver_parser=/tmp/cargo-installvGQBzF/release/deps/libsemver_parser-03a9543a582f3919.rmeta --extern serde=/tmp/cargo-installvGQBzF/release/deps/libserde-99864a521e391ccb.rmeta --extern serde_json=/tmp/cargo-installvGQBzF/release/deps/libserde_json-c1aabc2c6918fa37.rmeta --cap-lints allow`
     Running `/tmp/cargo-installvGQBzF/release/build/cargo-contract-b48c3d6c06a1f447/build-script-build`
memory allocation of 9897508864 bytes failed/148: binaryen-sys(build)                                                                                                                                                                                     
Aborted
```

**Comment by ascjones** [comment]
Yeah `0.8.0` comes with `binaryen-sys` bundled which requires it to be built, and it's failing there. We are looking to remove this so you can install it separately without having to build it as part of the installation. See https://github.com/paritytech/cargo-contract/issues/113

**Comment by cmichi** [comment]
@xhuanlee We've merged the PR #168 now, which enables you to install `cargo-contract` without the `binaryen` dependency. You need to install this dependency separately then, which should be possible without running into memory constraints.

Please see the [Readme](https://github.com/paritytech/cargo-contract#installation) for updated instructions on how to install `cargo-contract` without it.

This should fix your issues. If not, let us know please.
